queries/schedule.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will create or update BigQuery scheduled queries for each
+# of the derived tables.
+
+# Usage: 
+# `schedule.sh --query_file=<filename_of_source_query> \
+#  --table=<target_table> --project_id=<GCP_project_containing_dataset>`
+
+help() {
+    printf "Usage: schedule.sh --query_file=<filename_of_source_query> --table=<target_table> --project_id=<GCP_project_containing_dataset>\n"
+}
+
+# PARSE INPUTS
+for i in "$@"
+do
+case $i in
+    -q | --query_file ) QUERY_FILE="$2"; shift; shift ;;
+    -t | --table ) TABLE="$2"; shift; shift ;;
+    -p | --project_id ) PROJECT_ID="$2"; shift; shift ;;
+    -h | --help ) help; exit 0; shift; shift ;;
+    -- ) shift; break ;;
+    * ) break ;;
+  esac
+done
+
+if [[ -z "$QUERY_FILE" || -z "$TABLE" || -z "$PROJECT_ID" ]]
+then
+    printf "Error: one or more required arguments not specified\n"
+    help
+    exit 1
+fi
+
+echo QUERY_FILE=$QUERY_FILE
+echo TABLE=$TABLE
+echo PROJECT_ID=$PROJECT_ID
+
+# SCHEDULE THE QUERY
+
+# First, delete the transfer config, if it exists
+for location in US EU; do
+    while [ ! -z "$(bq ls --transfer_config --transfer_location=$location | grep "four_keys_$TABLE" -m 1 | awk '{print $1;}')" ]
```

Comment:
The previous display name was just the table name; eg,`changes` ... `four_keys_changes` is better, but if anyone was running the previous version, it won't find & delete the old `changes` schedules.  WDYT? Break with the past, in with the new?

Reply:
Hmm, you know, if someone were to use this script against an existing project, the behavior would be kind of gross: two scheduled tasks for each table, writing conflicting versions of the derived data.

So I undid that. Thanks for pointing it out!